DOC-3246: Suggested Edits and User Lookup API documentation (#3798)
tinymce/tinymce-docs: modules/ROOT/pages/userlookup.adoc

= Using the User Lookup API
:description: TinyMCE User Lookup provides an API to identify the current user and retrieve user details via integration-defined methods.
:keywords: userLookup, fetchUsers, userId, avatar, user metadata

The {productname} User Lookup API enables integrations to retrieve and cache user details (such as names and avatars) and identify the current user within the editor.

This is useful when building features that rely on user context, such as commenting, or displaying a list of elements container user information, and lets you manage user data efficiently with built-in caching.

== Using the User Lookup API

> Follow the steps below to configure and use the `userLookup` API.

Reviewer: ```suggestion
Follow the steps below to configure and use the `UserLookup` API.
```

> == Using the User Lookup API
> 
> Follow the steps below to configure and use the `userLookup` API.

Reviewer: This isn't using the API, it's implementing it for other code to use.
```suggestion
== Implementing the User Lookup API

Follow the steps below to configure the `UserLookup` API.
```

Reviewer 2: Removed in recent commit.

[[step-1-set-the-current-user]]

> === 1. Set the current user_id

Reviewer 2: Ill check over this

        )
      )
    )
  },
});
----

This example demonstrates how to simulate user lookup in-browser. In a production environment, this function could be swapped out for a network request.

The returned array should include objects with:

* `id` (string) - Required
* `name` (string) - Optional (defaults to `id`)
* `avatar` (string, URL or data URI) - Optional (auto-generated SVG fallback if missing)
* `custom` (object) - Optional metadata

[[step-3-fetch-user-information]]
=== 3. Fetch user information

Once the editor is initialized, use the API to look up multiple users:

.Example
[source,js]
----
editor.on('init', () => {
  const users = editor.userLookup.fetchUsers(['user-1', 'user-2']);

  users['user-1'].then((user) => {
    console.log('User 1 name:', user.name);
  });

  Promise.all(Object.values(users)).then((allUsers) => {
    allUsers.forEach((user) => {
      console.log(`Fetched: ${user.name} (${user.id})`);
    });
  });
});
----

NOTE: All lookups are cached. If a user has already been fetched, the same promise will be returned without calling `fetch_users` again.

[[step-4-handle-missing-or-default-users]]

> === 4. Handle default users if `fetch_users` is not provided

Reviewer: This is not actually a step. It's what happens if step 2 is not completed. But if it remains a step, it should at least be clear about what it contains.
```suggestion
=== 4. Defaults users if `fetch_users` is not provided
```

Reviewer 2: This has been resolved in recent update.

* `name` will default to the ID
* `avatar` will be a generated SVG using the first letter of the name

This is useful for simpler setups or prototyping.

.Example
[source,js]
----
tinymce.init({
  selector: '#editor',
  user_id: 'anonymous-guest'
});

editor.on('init', () => {
  const users = editor.userLookup.fetchUsers(['anonymous-guest']);
  users['anonymous-guest'].then((user) => {
    console.log('Generated fallback user:', user);
  });
});
----

[[step-5-integrate-user-lookup-in-a-ui]]
=== 5. Integrate User Lookup in a UI Button (Optional)

. Example: Using the API to show the current user's name via a toolbar button.
[source,js]
----
setup: (editor) => {
  editor.ui.registry.addButton('show-user', {
    text: 'Show User',
    onAction: () => {
      const currentUser = editor.userLookup.userId;
      const users = editor.userLookup.fetchUsers([currentUser]);

      users[currentUser].then((user) => {
        editor.notificationManager.open({
          text: `Logged in as: ${user.name}`,
          type: 'info'
        });
      });
    }
  });
}
----

Then add this to your toolbar:

[source,js]
----
toolbar: 'show-user'
----

[[summary]]

> == Summary

Reviewer: I don't know how this is done in our documentation format, but this should be an `API` section not a "summary" section. Something like how revision history did.
https://www.tiny.cloud/docs/tinymce/latest/revisionhistory/#apis
